packages/server/src/data/db/db-operations/post-ops.js

```diff
@@ -102,7 +104,77 @@ const getPostsByUserID = async (userId, sortType) => {
         ? { score: -1, date: -1, _id: -1 }
         : { date: -1, _id: -1 },
   });
-  return Post.aggregate(aggregation);
+  const posts = await Post.aggregate(aggregation);
+  return posts;
+};
+
+const getVotedPostsByUserID = async (userId, upvote) => {
+  const voteField = upvote ? "upvoters" : "downvoters";
+
+  const aggregation = [];
+  aggregation.push({
+    $match: {
+      [voteField]: { $in: [userId] },
+    },
+  });
+  aggregation.push({
+    $project: {
+      title: true,
+      body: true,
+      date: true,
+      numUpvotes: true,
+      numDownvotes: true,
+      username: true,
+      tag: true,
+      score: { $subtract: ["$numUpvotes", "$numDownvotes"] },
+      isMature: true,
```

Comment:
Basically can just add this:

isUpvoted: { $in: [userId, "$upvoters"] },
isDownvoted: { $in: [userId, "$downvoters"] },